### PR TITLE
Fix bugs in TorpedoStore.java

### PR DESCRIPTION
**Pull request by annab97**
Sonarlint found two major and one minor vulnarilibity in TorpedoStore.java.
- random generator must be inicialialized in constructor for guaranteed uniform distribution
-Exception was created but not throwed wich made the code vulnerable against out-of-range input values.
- when decreasing torpedocount =- was used instead of -= which left the model in an invalid state.

### CODE REVIEW

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 1]
```diff
@@ -12,10 +12,13 @@
   // rate of failing to fire torpedos [0.0, 1.0]
   private double FAILURE_RATE = 0.0; //NOSONAR
 
+  Random generator;
```
Comment (radl97) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java]: I would prefer a static Random, but this is also good.

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 2]
```diff
@@ -30,18 +33,17 @@ public TorpedoStore(int numberOfTorpedos){
 
   public boolean fire(int numberOfTorpedos){
     if(numberOfTorpedos < 1 || numberOfTorpedos > this.torpedoCount){
-      new IllegalArgumentException("numberOfTorpedos");
+      throw new IllegalArgumentException("numberOfTorpedos");
```
Comment (radl97) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:35]: ok

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 3]
```diff
@@ -30,18 +33,17 @@ public TorpedoStore(int numberOfTorpedos){
...
     boolean success = false;
 
     // simulate random overheating of the launcher bay which prevents firing
-    Random generator = new Random();
     double r = generator.nextDouble();
 
     if (r >= FAILURE_RATE) {
       // successful firing
-      this.torpedoCount =- numberOfTorpedos;
+      this.torpedoCount -= numberOfTorpedos;
```
Comment (radl97) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:45]: ok

**Review by radl97 — APPROVED** [review]
Good job :)

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java (hunk 1 above)
Comment (radl97) [reply src/main/java/hu/bme/mit/spaceship/TorpedoStore.java]: Can you make it static please?

**Review by radl97 — CHANGES_REQUESTED** [review]
Oh sry one more thing

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java (hunk 1 above)
Comment (annab97) [reply src/main/java/hu/bme/mit/spaceship/TorpedoStore.java]: OK, I will.

**Review by radl97 — APPROVED** [review]
Perfection! https://gph.is/2cMyrtE

**Pull request merged**